libs/jsonforms-components/src/lib/Controls/AddressLookup/AddressLookUpControl.tsx

```diff
@@ -112,40 +119,122 @@ export const AddressLookUpControl = (props: AddressLookUpProps): JSX.Element =>
     setSuggestions([]);
     setErrors({});
   };
+
+  /* istanbul ignore next */
   const handleRequiredFieldBlur = (name: string) => {
-    if ((!data?.[name] || data?.[name] === '') && requiredFields?.includes(name)) {
-      const err = { ...errors };
-      err[name] = name === 'municipality' ? `city is required` : `${name} is required`;
+    const err = { ...errors };
+    if(data?.["city"] === undefined || data?.["city"] === ""){
+      err[name] = name === 'municipality' ? 'city is required' : ""
       setErrors(err);
-    } else {
+    }
+
+    if(!data?.[name] || data[name] === '' || data?.[name] === undefined){
+      err[name] = name === 'municipality' ? 'city is required' : `${name} is required`;
+      setErrors(err);
+    }
+
+    if(!data?.[name]){
+      err[name] = name === 'addressLine1' ? `${name} is required` : ``;
+      setErrors(err);
+    }
+
+    else{
       delete errors[name];
     }
+
+    setTimeout(() => {
+      setSuggestions([])
+      setOpen(false)
+    }, 100);
   };
+
+  useEffect(() => {
+    if (dropdownRef.current) {
+        const selectedItem = dropdownRef.current.children[selectedIndex];
+        if (selectedItem) {
+          selectedItem.scrollIntoView({
+            block: 'nearest',
+            behavior: 'smooth',
+          });
+        }
+      }
+    }, [selectedIndex, open]);
+
+/* istanbul ignore next */
+const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'ArrowDown') {
+        e.preventDefault();
+        setSelectedIndex((prevIndex) =>
+            prevIndex < suggestions.length - 1 ? prevIndex + 1 : 0
+        );
+        handleDropdownChange(e.currentTarget.value)
+    } else if (e.key === 'ArrowUp') {
+        e.preventDefault();
+        setSelectedIndex((prevIndex) =>
+            prevIndex > 0 ? prevIndex - 1 : suggestions.length - 1
+        );
+        handleDropdownChange(e.currentTarget.value)
+    } else if (e.key === 'Enter') {
+      handleDropdownChange(e.currentTarget.value)
+      setLoading(false);
+      if (selectedIndex >= 0) {
+        document.getElementById("goaInput")?.blur()
+        const suggestion = suggestions[selectedIndex];
+        if(suggestion){
+          setTimeout(() => {
+              handleSuggestionClick(suggestion);
+              setOpen(false);
+            }, 100);
+          }
+        }
+    }
+};
+
   return (
     <div>
       {renderHelp()}
       <GoAFormItem label={label} error={errors?.['addressLine1'] ?? ''} data-testId="form-address-line1">
-        <SearchBox>
+        <SearchBox
+            onKeyDown={(e) => {
```

Comment:
The onKeyDown is causing the build error for jsonforms-ccomponents.
If you remove it and build it and be fine,

I think you should move the onkeydown to the GoAInput control as  i think you are trying to listen to key presses on the textbox.

You will have to add a event listener to the GoAInput control  on the useEffect.  See Dropdown.tsx for an example.